### ISSUE TITLE
is there any more detailed doc about pytorch_quantization?

### ISSUE THREAD

**Issue by ardeal** [NVIDIA/TensorRT#3340]
Hi,

I did see the doc: https://docs.nvidia.com/deeplearning/tensorrt/pytorch-quantization-toolkit/docs/userguide.html#

I am wondering whther there is some more detailed doc about:
1) how to create a new `relu`, `batchnorm2d`, and etc quant layers?
2) is the `quant_nn.Conv2d` a fake quant(float 8 type) layer rather than real quant(int 8)?
3) I would like to create a new fake quant layer `F.interpolate`, how to create it?
4) I would like to run the inference in `int8 ` value rather than fractional value, how to do it?

```python
from torch import nn

from pytorch_quantization import tensor_quant
import pytorch_quantization.nn as quant_nn

# pytorch's module
fc1 = nn.Linear(in_features, out_features, bias=True)
conv1 = nn.Conv2d(in_channels, out_channels, kernel_size)

# quantized version
quant_fc1 = quant_nn.Linear(
    in_features, out_features, bias=True,
    quant_desc_input=tensor_quant.QUANT_DESC_8BIT_PER_TENSOR,
    quant_desc_weight=tensor_quant.QUANT_DESC_8BIT_LINEAR_WEIGHT_PER_ROW)
quant_conv1 = quant_nn.Conv2d(
    in_channels, out_channels, kernel_size,
    quant_desc_input=tensor_quant.QUANT_DESC_8BIT_PER_TENSOR,
    quant_desc_weight=tensor_quant.QUANT_DESC_8BIT_CONV2D_WEIGHT_PER_CHANNEL)

```

**Comment by zerollzeng** [comment]
https://docs.nvidia.com/deeplearning/tensorrt/pytorch-quantization-toolkit/docs/tutorials/creating_custom_quantized_modules.html

Will it help? cc @ttyio

**Comment by ardeal** [comment]
@zerollzeng 
Thank you for your reply!
The link really helps, but it doesn't contained the information about those upper questions I asked.

The most import question is that:
I could train model with pytorch quantization in TensorRT, but I don't know how to run int8(not float8) inference after the training.

**Comment by zerollzeng** [comment]
> I could train model with pytorch quantization in TensorRT, but I don't know how to run int8(not float8) inference after the training.

You can refer to trtexec. a quick way is to run with 'trtexec --onnx=qat.onnx --int8'

**Comment by zerollzeng** [comment]
For python, you can refer to our polygraphy tools.

**Comment by ttyio** [comment]
@ardeal , 

> 1. how to create a new relu, batchnorm2d, and etc quant layers?

No need to do this, relue and BN are fused into conv, insert extra Q/DQ would break the fusion. more detail in https://docs.nvidia.com/deeplearning/tensorrt/developer-guide/index.html#qdq-placement-recs

> 2. is the quant_nn.Conv2d a fake quant(float 8 type) layer rather than real quant(int 8)?

Yes, fake quant with computation in FP32 during the fine-tuning with Pytorch. And we use computation in INT8 when deploy with TensorRT.

> 3. I would like to create a new fake quant layer F.interpolate, how to create it?

No need to do this because onnx `resize` commutes with Q/DQ. 

> 4. I would like to run the inference in int8  value rather than fractional value, how to do it?

As @zerollzeng  said, TRT use INT8.

**Comment by ttyio** [comment]
closing since no activity for more than 3 weeks, pls reopen if you still have question, thanks!
